examples/e2e/steps/alerts/slack_notifications.py

```diff
@@ -0,0 +1,53 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+
+from config import PipelinesConfig
+
+from zenml import get_step_context
+from zenml.client import Client
+from zenml.utils.dashboard_utils import get_run_url
+
+alerter = Client().active_stack.alerter
+
+
+def build_slack_message(status: str) -> str:
+    """Builds a message for Slack post.
+
+    Args:
+        status: Status to be set in text.
+
+    Returns:
+        str: Prepared message.
+    """
+    step_context = get_step_context()
+    run_url = get_run_url(step_context.pipeline_run)
+
+    return (
+        f"Pipeline `{step_context.pipeline_name}` [{str(step_context.pipeline.id)}] {status}!\n"
+        f"Run `{step_context.pipeline_run.name}` [{str(step_context.pipeline_run.id)}]\n"
+        f"URL: {run_url}"
+    )
```

Comment:
I think it might make sense to add this into the Slack alerter by default. We already have a `include_format_blocks` option, which, when set, prettifies the message you send via markdown and that already includes the pipeline and step names. Adding run and run URL might make sense there IMO

Reply:
This will not work on just `Alerter` isn't it?  I'm relying on stack info to get alerter and fire plain `post` to it.

Btw, a side note: why do we allow only one alerter in the stack? People often want email and Slack (or other means), maybe also with some flexibility on recipient lists.